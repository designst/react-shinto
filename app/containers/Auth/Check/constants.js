import createConstant from 'utils/createConstant';

export const AUTH_CHECK_REQUEST = createConstant(__filename, 'AUTH_CHECK_REQUEST');
export const AUTH_CHECK_SUCCESS = createConstant(__filename, 'AUTH_CHECK_SUCCESS');
export const AUTH_CHECK_FAILURE = createConstant(__filename, 'AUTH_CHECK_FAILURE');
