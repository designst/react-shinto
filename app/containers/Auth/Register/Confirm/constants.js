import createConstant from 'utils/createConstant';

export const AUTH_REGISTER_CONFIRM_REQUEST = createConstant(
  __filename,
  'AUTH_REGISTER_CONFIRM_REQUEST',
);
export const AUTH_REGISTER_CONFIRM_SUCCESS = createConstant(
  __filename,
  'AUTH_REGISTER_CONFIRM_SUCCESS',
);
export const AUTH_REGISTER_CONFIRM_FAILURE = createConstant(
  __filename,
  'AUTH_REGISTER_CONFIRM_FAILURE',
);
