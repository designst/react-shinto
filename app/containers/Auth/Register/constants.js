import createConstant from 'utils/createConstant';

export const AUTH_REGISTER_REQUEST = createConstant(__filename, 'AUTH_REGISTER_REQUEST');
export const AUTH_REGISTER_SUCCESS = createConstant(__filename, 'AUTH_REGISTER_SUCCESS');
export const AUTH_REGISTER_FAILURE = createConstant(__filename, 'AUTH_REGISTER_FAILURE');
