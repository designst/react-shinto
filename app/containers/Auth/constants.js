import createConstant from 'utils/createConstant';

export const AUTH_API_REQUEST = createConstant(__filename, 'AUTH_API_REQUEST');
export const AUTH_API_SUCCESS = createConstant(__filename, 'AUTH_API_SUCCESS');
export const AUTH_API_FAILURE = createConstant(__filename, 'AUTH_API_FAILURE');
