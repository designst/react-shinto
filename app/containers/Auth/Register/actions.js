import { AUTH_REGISTER_REQUEST, AUTH_REGISTER_SUCCESS, AUTH_REGISTER_FAILURE } from './constants';

export const registerRequest = payload => ({
  type: AUTH_REGISTER_REQUEST,
  payload,
});

export const registerSuccess = data => ({
  type: AUTH_REGISTER_SUCCESS,
  data,
});

export const registerFailure = error => ({
  type: AUTH_REGISTER_FAILURE,
  error,
});
