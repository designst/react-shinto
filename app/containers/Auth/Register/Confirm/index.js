import { compose } from 'redux';
import { connect } from 'react-redux';

import RegisterConfirm from 'components/Auth/Register/Confirm';

import { registerConfirmRequest } from './actions';

const mapStateToProps = (state, { match }) => ({
  token: match.params.token,
});

const mapDispatchToProps = {
  onSubmit: registerConfirmRequest,
};

const connector = connect(
  mapStateToProps,
  mapDispatchToProps,
);

export default compose(connector)(RegisterConfirm);
