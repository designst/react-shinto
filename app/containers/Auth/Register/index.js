import { compose } from 'redux';
import { connect } from 'react-redux';

import Register from 'components/Auth/Register';

import { registerRequest } from './actions';

const mapStateToProps = () => ({});

const mapDispatchToProps = {
  onSubmit: registerRequest,
};

const connector = connect(mapStateToProps, mapDispatchToProps);

export default compose(connector)(Register);
