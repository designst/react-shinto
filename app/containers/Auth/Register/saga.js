import { call, put, takeLatest, getContext } from 'redux-saga/effects';

import createLogger from 'utils/createLogger';

import { AUTH_REGISTER_REQUEST } from 'containers/Auth/Register/constants';
import { registerSuccess, registerFailure } from 'containers/Auth/Register/actions';

const logger = createLogger(__filename);

const authRegisterUrl = process.env.SHINTO_AUTH_REGISTER_API_ENDPOINT;

export function* authRegister(action) {
  const { email, username, password, passwordConfirm } = action;

  const apiService = yield getContext('apiService');

  logger('Register: %s', email);

  try {
    const response = yield call(apiService.post, authRegisterUrl, {
      email,
      username,
      password,
      passwordConfirm,
    });
    yield put(registerSuccess(response.data));
  } catch (err) {
    yield put(registerFailure(err));
  }
}

export default function* authRegisterData() {
  yield takeLatest(AUTH_REGISTER_REQUEST, authRegister);
}
