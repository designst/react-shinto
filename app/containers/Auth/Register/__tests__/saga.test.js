import { expectSaga } from 'redux-saga-test-plan';

import authReducer from 'containers/Auth/reducer';
import authRegisterData from 'containers/Auth/Register/saga';
import { AUTH_REGISTER_REQUEST } from 'containers/Auth/Register/constants';

describe('authRegisterData', () => {
  let authRegisterGenerator;

  beforeEach(() => {
    authRegisterGenerator = authRegisterData();

    const selectDescriptor = authRegisterGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();
  });

  it('should dispatch the authRegisterSuccess action if it requests the data successfully', () => {
    const token = 'token';

    const email = 'email@email.com';
    const username = 'username';
    const password = 'password';

    return expectSaga(authRegisterData)
      .provide({
        getContext: () => ({
          post: url =>
            url === '/auth/register'
              ? {
                  data: {
                    token,
                    email,
                    username,
                  },
                }
              : {},
        }),
      })
      .dispatch({
        type: AUTH_REGISTER_REQUEST,
        payload: {
          email,
          username,
          password,
          passwordConfirm: password,
        },
      })
      .withReducer(authReducer)
      .run();
  });

  it('should dispatch the authRegisterFailure action if it requests the data unsuccessfully', () => {
    const email = 'email@email.com';
    const username = 'username';
    const password = 'password';

    return expectSaga(authRegisterData)
      .dispatch({
        type: AUTH_REGISTER_REQUEST,
        payload: {
          email,
          username,
          password,
          passwordConfirm: password,
        },
      })
      .withReducer(authReducer)
      .run();
  });
});
