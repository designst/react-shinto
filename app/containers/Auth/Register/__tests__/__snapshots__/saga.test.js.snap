// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`authRegisterData should dispatch the authRegisterFailure action if it requests the data unsuccessfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "FORK": Object {
    "args": Array [
      "@shinto/app/containers/auth/register/constants/AUTH_REGISTER_REQUEST",
      [Function],
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`authRegisterData should dispatch the authRegisterSuccess action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "FORK": Object {
    "args": Array [
      "@shinto/app/containers/auth/register/constants/AUTH_REGISTER_REQUEST",
      [Function],
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
